Fix logical flow between the relying party components and other misc fixes

bpradipt opened the pull request:
Fixes: #105

bpradipt committed:
Update architecture doc and diagrams

The patch includes number of fixes for the architecture doc.
Fixes the logical flow between the attestation agent and relying party
for all the diagrams.
Fixes the architecture diagram for process-based TEEs and replaces
references to inclavare with enclave-cc.
Added the architecture diagram for peer-pods approach.
Finally updated the markdown to use relative paths for the images to make
it easier for viewing during reviews and editors.

Signed-off-by: Pradipta Banerjee <pradipta.banerjee@gmail.com>

mythi commented:
I started updating related docs in this repo and here's what I got:

```
diff --git a/architecture.md b/architecture.md
index bff2622..e6d17fa 100644
--- a/architecture.md
+++ b/architecture.md
@@ -196,8 +196,6 @@ Containers using VM-based TEEs and the Kata Containers runtime:
 ![COCO_ccv1_TEE](https://github.com/confidential-containers/documentation/blob/main/images/COCO_ccv1_TEE.jpg)
 
 The following diagram shows the upcoming v1 architecture to run Confidential
-Containers using the Intel SGX process-based TEE. It relies on the
-[Inclavare Containers](https://github.com/inclavare-containers/inclavare-containers)
-runtime:
+Containers using the Intel SGX process-based TEE:
```

Since you're updating the arch doc, maybe drop Inclavare from the diagrams too?

We don't rely on `rune`. We have a custom shim called `shim-rune` but it uses `runc`.

dcmiddle reviewed (approved):
Thanks for the fix!

fitzthum commented on images/COCO_ccv1_TEE.png:
I think both arrows to/from the Attestation Agent should go to the KBS. The Attestation Service never talks to the AA directly.

dcmiddle replied:
Oh true. I should have caught that.

bpradipt replied:
Right.. I'll fix it.

bpradipt commented:
I updated the doc and all the figures, including the process-based TEE diagram and references. Also added a diagram for peer-pods approach.

@dcmiddle @sameo @mythi @fitzthum ptal

mythi reviewed (approved):
enclave-cc LGTM, thanks!
